### ISSUE TITLE
Improve pipeline (tobi)

### ISSUE THREAD

**Issue by tfiedlerdev** [LeonHermann322/hot-prot#14]
A- [ ] check pipeline (See if just using the average as the summarizer of the per_repr_pos yields the same results as without summarizer)
- [x] do dataset split using UniRef50 clusters instead of FLIP clustering (this way we might not waste as many samples)--> less overfitting??
Before with FLIP clustering
![image](https://user-images.githubusercontent.com/29177177/221027183-f60154b8-5458-4828-9b59-e3b8602ea257.png)
After with UniRef clustering
![image](https://user-images.githubusercontent.com/29177177/221026671-ff8d17dd-3b90-4ead-a752-83e242964045.png)

- [x] add test set to split
- [x] implement early stopping (if overfitting or non improvement of val loss detected over e.g. 3 epochs) --> less time wasted
